test/puppetlabs/puppetdb/cli/services_test.clj

```diff
@@ -506,21 +506,38 @@
                         #"FATAL: terminating connection due to administrator command"
                         (deref resource-query default-timeout-ms :timeout-getting-expected-query-ex))))
 
-                 ;; There should be at least two cancelled messages in
-                 ;; the logs. One from the queries cancelled by report
-                 ;; GC and one from resource_events GC.  There may be
-                 ;; more if any of the SIGTERMed pg workers don't exit
-                 ;; before the bulldozer runs again.
-                 (is (<= 2 (->> @log
-                                (map :message)
-                                (map #(str/includes? % "Partition GC terminated queries"))
-                                (filter true?)
-                                count)))
-
-                 (is (= (->> report-parts (sort-by :table) (drop 1) set)
-                        (set (get-temporal-partitions "reports"))))
-                 (is (= (->> event-parts (sort-by :table) (drop 1) set)
-                        (set (get-temporal-partitions "resource_events")))))))))))))
+                 (let [expected-report-parts (->> report-parts (sort-by :table) (drop 2) set)
+                       expected-event-parts (->> event-parts (sort-by :table) (drop 2) set)]
+
+                   ;; There is some delay between garbage collection completing
```

Comment:
Could do something like this to pass. But should garbage collection have already waited on this?

Reply:
Going over this with Austin, he pointed out that gc drops one table per run, so the test expectation is correct. So in ad30d43, to avoid the race, I added a before-gc cyclicbarrier to block next gc execution after after-gc point is reached.